src/coreComponents/codingUtilities/UnitTestUtilities.hpp

```diff
@@ -87,14 +87,14 @@ ::testing::AssertionResult checkRelativeErrorFormat( const char *, const char *,
 {
   real64 const delta = std::abs( v1 - v2 );
   real64 const value = std::max( std::abs( v1 ), std::abs( v2 ) );
-  if( delta > absTol && delta > relTol * value )
+  if( delta > absTol && delta > relTol * (value + 1.0) )
```

Comment:
@joshua-white @klevzoff is it ok like that?

Reply:
Do you need both conditions?  Personally I would drop the `absTol` one.  In the situation where `value` is very large in magnitude, a large magnitude `delta` may still be indicative of a good solution.  I'd like to hear @klevzoff's opinion though.

Reply:
`absTol` was meant to allow us to ignore benign diffs below machine precision when the values are also small in magnitude (in this case `relTol` often fails). Not sure about the default value though, seems like it should've been 0, i.e. strict checking by default?

The proposed change can introduce false negatives for values << 1, for example
```
value = 1e-3
delta = 1e-8
relTol = 1e-6
absTol = 1e-14
```
would previously fail the test (`1e-8 > 1e-9 == true`), but will now pass (`1e-8 > 1.001e-6 == false`).

Maybe we just need more flexibility w.r.t. customizing tolerances in tests? I.e. instead of using the same `absTol` and `relTol` for all numerical derivative checking, we should be specifying them separately for pressure vs density and separately for each property/derivative tested, on a per-model basis. Then we could use `absTol` to simply ignore spurious diffs like the one @paveltomin is dealing with. Does this seem implementable?

Reply:
I think the following two conditions
```
delta > absTol
&&
delta > relTol*value
```
should be essentially equivalent to the single condition
```
delta > relTol * (value + absTol / relTol)
```
One is simply a smoothed version of the other.  Using @klevzoff's example values, we would pass the same test.  Instead choosing 1.0 for the last coefficient implies `absTol = relTol`.

So maybe either implementation is fine.  I think the core issue was that the chosen tolerances are stringent.  In the finite difference test, the derivative estimation is only O(h), where h is something like `sqrt(epsilon)`.  We currently have `relTol = epsilon`.  Perhaps we should switch to the centered difference approximation, which is O(h^2) and we can maybe keep stringent tolerances?

Reply:
i just want to merge those bug fixes to be honest
not that much interested in dealing with unit tests